encoders/firebase-encoders-json/src/main/java/com/google/firebase/encoders/FieldDescriptor.java

```diff
@@ -62,9 +64,13 @@ public String getName() {
    * @return {@code T} annotation if present, null otherwise.
    */
   @Nullable
-  @SuppressWarnings("unchecked")
   public <T extends Annotation> T getProperty(@NonNull Class<T> type) {
-    return (T) properties.get(type);
+    return type.cast(properties.get(type));
+  }
+
+  @NonNull
+  public Set<Annotation> getAllAnnotations() {
+    return new HashSet<>(properties.values());
```

Comment:
Instead of copying, consider using `Collections.unmodifiableSet(properties.values())`, this would prevent modification and avoid a full copy.

Reply:
furthermore, I think it would be even cleaner if you stored the properties above as `this.properties = Collections.unmodifiableMap(properties)`. then you don't need to allocate a new unmodifiableSet here at all, just `return properties.values()`